core/src/main/scala/org/apache/spark/SparkException.scala

```diff
@@ -30,3 +30,13 @@ class SparkException(message: String, cause: Throwable)
  */
 private[spark] class SparkDriverExecutionException(cause: Throwable)
   extends SparkException("Execution error", cause)
+
+/**
+ * Exception indicating an error internal to Spark -- it is in an inconsistent state, not due
+ * to any error by the user
+ */
+class SparkIllegalStateException(message: String, cause: Throwable)
```

Comment:
Why not use the existing IllegalStateException (the non-spark specific one)? That's what done elsewhere in Spark.